lib/Twig/Template.php

```diff
@@ -205,7 +205,11 @@ public function displayBlock($name, array $context, array $blocks = array(), $us
         } elseif (false !== $parent = $this->getParent($context)) {
             $parent->displayBlock($name, $context, array_merge($this->blocks, $blocks), false);
         } else {
-            throw new Twig_Error_Runtime(sprintf('Block "%s" on template "%s" does not exist.', $name, $this->getTemplateName()), -1, $this->getTemplateName());
+            if (isset($blocks[$name])) {
```

Comment:
this could be an `elseif` branch of the surrounding condition thus avoiding another level of nesting